BlueShift-iOS-SDK/BlueshiftEventAnalyticsHelper.m

```diff
@@ -15,53 +15,56 @@
+    if (pushDetailsDictionary) {
+        NSString *bsft_experiment_uuid = [self getValueBykey: pushDetailsDictionary andKey: kInAppNotificationModalExperimentIDKey];
+        NSString *bsft_user_uuid = [self getValueBykey: pushDetailsDictionary andKey: kInAppNotificationModalUserIDKey];
+        NSString *message_uuid = [self getValueBykey: pushDetailsDictionary andKey: kInAppNotificationModalMessageUDIDKey];
+        NSString *transactional_uuid = [self getValueBykey: pushDetailsDictionary andKey: kInAppNotificationModalTransactionIDKey];
+        NSString *sdkVersion = [NSString stringWithFormat:@"%@", kSDKVersionNumber];
+        NSString *clickElement = [self getValueBykey: pushDetailsDictionary andKey: kNotificationClickElementKey];
+        NSString *urlElement = [self getValueBykey: pushDetailsDictionary andKey: kNotificationURLElementKey];
+        NSString *deviceId = [[BlueShiftDeviceData currentDeviceData] deviceUUID];
+        NSString *appName = [[BlueShiftAppData currentAppData] bundleIdentifier];
+        NSString *pushDeepLinkURL = [self getValueBykey: pushDetailsDictionary andKey: kPushNotificationDeepLinkURLKey];
+        NSString *timestamp = [self getCurrentUTCTimestamp];
+        if (bsft_user_uuid) {
+            [pushTrackParametersMutableDictionary setObject:bsft_user_uuid forKey: kInAppNotificationModalUIDKey];
+        }
+        if(bsft_experiment_uuid) {
+            [pushTrackParametersMutableDictionary setObject:bsft_experiment_uuid forKey: kInAppNotificationModalEIDKey];
+        }
+        if (message_uuid) {
+            [pushTrackParametersMutableDictionary setObject:message_uuid forKey: kInAppNotificationModalMIDKey];
+        }
+        if (transactional_uuid) {
+            [pushTrackParametersMutableDictionary setObject:transactional_uuid forKey: kInAppNotificationModalTXNIDKey];
+        }
+        if (sdkVersion) {
+            [pushTrackParametersMutableDictionary setObject:sdkVersion forKey: kInAppNotificationModalSDKVersionKey];
+        }
+        if ([self isNotNilAndNotEmpty:clickElement]) {
+            [pushTrackParametersMutableDictionary setObject:clickElement forKey: kNotificationClickElementKey];
+        }
+        if ([self isNotNilAndNotEmpty:urlElement]) {
+            [pushTrackParametersMutableDictionary setObject:urlElement forKey: kNotificationURLElementKey];
+        }
+        if([[pushDetailsDictionary objectForKey: kNotificationTypeIdentifierKey] isEqualToString:kNotificationKey] && [self isNotNilAndNotEmpty:pushDeepLinkURL]) {
+            NSString *encodedUrl = [BlueShiftInAppNotificationHelper getEncodedURLString:pushDeepLinkURL];
+            if (encodedUrl) {
+                [pushTrackParametersMutableDictionary setObject:encodedUrl forKey: kNotificationURLElementKey];
+            }
+        }
+        if (deviceId) {
+            [pushTrackParametersMutableDictionary setObject:deviceId forKey: @"device_id"];
```

Comment:
I saw you adding constants for these two strings in a previous PR. can we use them here?

Reply:
for device_id and app_name

Reply:
fixed.